File: fixtures/conformance/ruby/lib/store.rb
require 'json'
require_relative 'base'
require_relative 'loggable'

module Acme
  class Store < Base
    include Loggable
    extend Enumerable

    VERSION = "1.0"
    DEFAULTS = { ttl: 60 }

    attr_accessor :name

    def initialize(name)
      @name = name
    end

    def find(key)
      log(key)
      JSON.parse(key)
    end

    def get(key)
      find(key)
    end

    alias fetch get

    def self.create(name)
      new(name)
    end

    class << self
      def reset
        create("default")
      end
    end
  end
end

File: fixtures/conformance/php/src/Store.php
<?php

namespace App\Store;

use App\Support\Logger;
include 'legacy.php';

final class Store extends Base implements Readable, \JsonSerializable
{
    use Cacheable;

    public const VERSION = '1.0';

    private string $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function find(string $key): string
    {
        Logger::write($key);
        return $this->data[$key] ?? '';
    }

    #[\Override]
    public function read(string $key): string
    {
        return $this->find($key);
    }

    public static function create(string $name): self
    {
        return new self($name);
    }
}

function build(): Store
{
    return new Store('x');
}
